src/data/file_preprocessing/echo_files_parser.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+import pandas as pd
+
+class EchoFilesParser:
+    def __init__(
+        self,
+        echo_files: list[str],
+    ):
+        """
+        :param echo_files: list of echo file names
+        """
+        self.echo_files = echo_files
+        self.echo_df = pd.DataFrame()
+        self.exceptions_df = pd.DataFrame()
+
+    def find_marker_rows(self, file:str, markers:tuple[str]) -> list[int]:
```

Comment:
small styling issue: there should be a whitespace between param name and type hint (i.e. `file: str`)
probably we should configure our linting to do that

Reply:
Done, `black` came handy with that ;)